src/backend/joanie/tests/core/test_api_admin_contract_definitions.py

```diff
@@ -0,0 +1,160 @@
+            content_type="application/json",
+            data=data,
+        )
+
+        self.assertEqual(response.status_code, 201)
+        content = response.json()
+        self.assertIsNotNone(content["id"])
+        self.assertEqual(content["title"], "Contract 001")
+
+    def test_admin_api_contract_definition_update(self):
+        """
+        Staff user should be able to update a contract definition.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        certification_definition = factories.ContractDefinitionFactory(
+            title="Contract grade 1", name=enums.CONTRACT_NAME_CHOICES[0][0]
+        )
+        payload = {
+            "title": "Updated Contract grade 1",
+            "name": enums.CONTRACT_NAME_CHOICES[0][0],
+            "language": "fr-fr",
+        }
+
+        response = self.client.put(
+            f"/api/v1.0/admin/contract-definitions/{certification_definition.id}/",
+            content_type="application/json",
+            data=payload,
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = response.json()
+        self.assertEqual(content["id"], str(certification_definition.id))
+        self.assertEqual(content["title"], "Updated Contract grade 1")
+        self.assertEqual(content["name"], "contract_definition")
+
+    def test_admin_api_contract_definition_partially_update(self):
+        """
+        Staff user should be able to partially update a certification_definition.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        certification_definition = factories.ContractDefinitionFactory(
+            name=enums.CONTRACT_DEFINITION
+        )
+
+        response = self.client.patch(
+            f"/api/v1.0/admin/contract-definitions/{certification_definition.id}/",
+            content_type="application/json",
+            data={"name": enums.CONTRACT_DEFINITION},
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = response.json()
+        self.assertEqual(content["id"], str(certification_definition.id))
+        self.assertEqual(content["name"], "contract_definition")
+
+    def test_admin_api_contract_definition_delete(self):
+        """
+        Staff user should be able to delete a contract definition.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        contract_definition = factories.ContractDefinitionFactory()
+
+        response = self.client.delete(
+            f"/api/v1.0/admin/contract-definitions/{contract_definition.id}/"
+        )
+
+        self.assertEqual(response.status_code, 204)
+        self.assertFalse(models.ContractDefinition.objects.exists())
```

Comment:
`create`, `update`, `partial_update` and `retrieve` permissions are not tested. Are you considering than the permissions are the same for all the actions ? How do you ensure that this is never changing ?

Reply:
I copy/pasted tests from the certificate endpoint. I would say tests on the admin endpoints have been kept minimal because we are relying on Django permission classes. I agree this should be improved but maybe in a separate PR so it is done accross all endpoints?